src/clarity-angular/tabs/vtabs.ts

```diff
@@ -0,0 +1,212 @@
+/*
+ * Copyright (c) 2016 - 2017 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import {Component, Input, Output, AfterContentInit, ContentChildren, QueryList, EventEmitter} from "@angular/core";
+
+import {VTabsNavLink} from "./vtabs-nav-link";
+import {VTabsContent} from "./vtabs-content";
+
+let nbVTabLinks: number = 0;
+
+/**
+ * @class VTabs
+ *
+ * @description
+ * This Component represents the container for a clr-vtabs element.
+ * Proper use requires users to pass in clr-vtabs-nav-group(s) or clr-vtabs-nav-links to set the navigation elements.
+ * Corresponding clr-vtabs-content items should map in number and position to the activation link.
+ *
+ * It always sets the class of .vtabs on its host.
+ * It exposes the @Input "clrReversed" which defaults to false and can be used to reverse the content and nav elements.
+ */
+@Component({
+    selector: "clr-vtabs",
+    template: `
+            <div class="vtabs-nav">
+                <ng-content select="clr-vtabs-nav-group"></ng-content>
+                <clr-vtabs-nav-group>
+                    <ng-content ngProjectAs="clr-vtabs-nav-link" select="clr-vtabs-nav-link"></ng-content>
+                </clr-vtabs-nav-group>
+            </div>
+            <ng-content select="clr-vtabs-content"></ng-content>
+    `,
+    host: {
+        "[class.vtabs]": "true",
+        "[class.is-reversed]": "reversed"
+    },
+    providers: []
+})
+export class VTabs implements AfterContentInit {
+
+    /**
+     * @property vtabLinks
+     *
+     * @description
+     * A QueryList of all VTabNavLink items.
+     */
+    @ContentChildren(VTabsNavLink, {descendants: true}) vtabLinks: QueryList<VTabsNavLink>;
+
+    /**
+     * @property vtabContents
+     *
+     * @description
+     * A QueryList of all VTabContent items.
+     */
+    @ContentChildren(VTabsContent, {descendants: true}) vtabContents: QueryList<VTabsContent>;
+
+    /**
+     * @description
+     * Input binding that allows the navigation to be on left or right hand side. Left side is default.
+     *
+     * @type {boolean}
+     */
+    @Input("clrReversed") public reversed: boolean = false;
+
+    /**
+     * @property linkChange
+     *
+     * @description
+     * EventEmitter for the tab link change event.
+     *
+     * @type {EventEmitter<VTabsNavLink>}
+     */
+    @Output("clrVTabsNavChange") linkChange: EventEmitter<VTabsNavLink> = new EventEmitter<VTabsNavLink>(false);
```

Comment:
It's safer, we explicitly state we want them synchronous. We've been burnt once when the Angular team made them asynchronous by default without announcing it.